src/CoherenceBuild/PackagePublisher.cs

```diff
@@ -59,15 +70,32 @@ public static void PublishToFeed(ProcessResult processResult, string feed, strin
                 int attempt = 0;
                 Program.Retry(() =>
                 {
+                    if (package.IsCoreCLRPackage && IsAlreadyUploaded(v3Feed, httpClient, package.Package))
+                    {
+                        Log.WriteInformation($"Skipping {package.Package} since it is already published.");
+                        return;
+                    }
+
                     attempt++;
-                    Log.WriteInformation($"Attempting to publish package {package.Package} ({attempt})");
+                    Log.WriteInformation($"Attempting to publish package {package.Package} (Attempt: {attempt})");
                     var length = new FileInfo(package.PackagePath).Length;
                     server.PushPackage(apiKey, new PushLocalPackage(package.PackagePath), length, (int)TimeSpan.FromMinutes(5).TotalMilliseconds, disableBuffering: false);
+                    Log.WriteInformation($"Done publishing package {package.Package}");
                 });
-                Log.WriteInformation($"Done publishing package {package.Package}");
             });
         }
 
+        private static bool IsAlreadyUploaded(string v3Feed, System.Net.Http.HttpClient client, IPackage package)
+        {
+            // https://myget-2e16.kxcdn.com/artifacts/aspnetvolatiledev/nuget/v3/flatcontainer/microsoft.aspnetcore.signalr.server/0.1.0-rc2-20311/microsoft.aspnetcore.signalr.server.0.1.0-rc2-20311.nupkg
+            var id = package.Id.ToLowerInvariant();
+            var version = package.Version.ToNormalizedString();
+            var uri = $"{v3Feed.TrimEnd('/')}/{id}/{version}/{id}.{version}.nupkg";
+            var message = new HttpRequestMessage(HttpMethod.Head, uri);
+            var result = client.SendAsync(message, HttpCompletionOption.ResponseHeadersRead).Result;
+            return result.IsSuccessStatusCode;
```

Comment:
Should you verify what response you get? I assume you only want to not upload it if you get a 404. A 500 for example would be different.